src/cdi_CPEloss/Tabular_CP_Eloss.cc

```diff
@@ -0,0 +1,246 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!                                                                              
+ * \file   cdi_CPEloss/Tabular_CP_Eloss.cc                                     
+ * \author Ben R. Ryan                                                        
+ * \date   2019 Nov 4                                             
+ * \brief  Tabular_CP_Eloss member definitions.                                 
+ * \note   Copyright (C) 2016-2019 Triad National Security, LLC.                 
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#include "Tabular_CP_Eloss.hh"
+
+using std::stod;
+using std::stoi;
+
+namespace rtt_cdi_cpeloss {
+
+//---------------------------------------------------------------------------//
+// CONSTRUCTORS
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Constructor for an analytic tabular eloss model. 
+ *
+ * This constructor builds an eloss model defined by the
+ * rtt_cdi_cpeloss::Tabular_Eloss_Model derived class argument.
+ *
+ * The path to an eloss datafile is passed to the constructor,
+ * which opens and parses the file. The file format is the usual
+ * LANL format for stopping powers. 
+ *
+ * \param filename_in path to eloss file
+ * \param target_zaid_in target particle zaid
+ * \param projectile_zaid_in transporting particle zaid
+ */
+Tabular_CP_Eloss::Tabular_CP_Eloss(std::string filename_in,
+                                   rtt_cdi::CParticle &target_in,
+                                   rtt_cdi::CParticle &projectile_in)
+    : filename(filename_in), rtt_cdi::CPEloss(target_in, projectile_in) {
```

Comment:
I think the base class constructor call should always come first in the initializer list, because it always gets called first in practice...

Reply:
Already fixed this and the next related issue on my local branch (they generate compiler warnings that I noticed too late), but hadn't pushed yet to avoid spamming everyone with "new commit" emails. Good catch though!